### ISSUE TITLE
linter: react/jsx-no-literals misses text literals inside a JSXExpressionContainer used as a JSX child

### ISSUE THREAD

**Issue by vigilosapp** [oxc-project/oxc#26208]
**What version of Oxlint are you using?**
1.80.0

**What command did you run?**
`oxlint src/repro.tsx`

**What does your `.oxlintrc.json` config file look like?**
```jsonc
{
  "plugins": ["react"],
  "rules": {
    "react/jsx-no-literals": "error"
  }
}
```

**What happened?**

`react/jsx-no-literals` misses an unwrapped text literal whenever the JSX
containing it is placed inside a `JSXExpressionContainer` (`{...}`) as the
child of another JSX element. This happens regardless of whether the
container holds a conditional expression (`&&`, ternary) or just a plain
JSX element with nothing conditional at all. The same literal is detected
when it's not behind a `JSXExpressionContainer` child, whether that's a
component's direct return value or a plain (non-`{}`) nested JSX child.

```tsx
// Detected: conditional expression is the component's direct
// return value, not inside a JSXExpressionContainer.
export function Direct({ ok }: { ok: boolean }) {
  return ok && <div>Text</div>
}

// NOT detected: identical conditional expression, now inside a
// JSXExpressionContainer as a child of <div>.
export function Nested({ ok }: { ok: boolean }) {
  return <div>{ok && <div>Text</div>}</div>
}

// Detected: plain nesting, no JSXExpressionContainer at all.
export function PlainNested() {
  return <div><span>Text</span></div>
}

// NOT detected: ternary instead of &&, same JSXExpressionContainer
// pattern as Nested.
export function Ternary({ ok }: { ok: boolean }) {
  return <div>{ok ? <span>Text</span> : null}</div>
}

// NOT detected: no conditional expression anywhere, just a static JSX
// element wrapped in a JSXExpressionContainer as a child of <div>. This
// rules out the conditional expression as the cause.
export function PlainContainer() {
  return <div>{<span>Text</span>}</div>
}
```

Running `oxlint src/repro.tsx` with the config above reports exactly two
violations — the `Text` literal in `Direct` and in `PlainNested`. `Nested`,
`Ternary`, and `PlainContainer` report nothing, even though all three
contain the same unwrapped `Text` literal.

This isn't limited to the default options: enabling `noStrings: true`
together with `ignoreProps: true` (to keep JSX attribute literals like
`data-testid="x"` out of scope) produces the identical result across all
five functions.

**Parity check against the reference implementation**

I ran the same file through ESLint 9.39.5 with `eslint-plugin-react`
7.37.5 (`react/jsx-no-literals: "error"`, no other options), the rule
this was ported from. It reports a violation for **all five** functions,
including the three oxlint misses:

| Function | oxlint 1.80.0 | eslint-plugin-react 7.37.5 |
| --- | --- | --- |
| `Direct` | flagged | flagged |
| `Nested` | not flagged | flagged |
| `PlainNested` | flagged | flagged |
| `Ternary` | not flagged | flagged |
| `PlainContainer` | not flagged | flagged |

So this isn't a case of oxlint documenting different behavior from the
rule it ported — the reference implementation flags all five, oxlint only
two.

Expected: `Nested`, `Ternary`, and `PlainContainer` should also report a
violation for `Text`, matching `eslint-plugin-react`.
